docs/source/system-metrics/index.rst

```diff
@@ -0,0 +1,214 @@
+single run. To do so, set ``log_system_metrics`` as ``True`` or ``False`` accordingly in :py:func:`mlflow.start_run()`:
+
+.. code-block:: python
+
+    with mlflow.start_run(log_system_metrics=True) as run:
+        time.sleep(15)
+
+    print(mlflow.MlflowClient().get_run(run.info.run_id).data)
+
+The above code will log system metrics for the current run, even if you have disabled system metrics logging by setting
+``MLFLOW_ENABLE_SYSTEM_METRICS_LOGGING`` to ``false`` or calling ``mlflow.disable_system_metrics_logging()``.
+
+Type of System Metrics
+----------------------
+
+By default MLflow logs the following system metrics:
+
+* cpu_utilization_percentage
+* system_memory_usage_megabytes
+* system_memory_usage_percentage
+* gpu_utilization_percentage
+* gpu_memory_usage_megabytes
+* gpu_memory_usage_percentage
+* network_receive_megabytes
+* network_transmit_megabytes
+* disk_usage_megabytes
+* disk_available_megabytes
+
+GPU metrics are only logged when GPU is available and ``pynvml`` is installed.
+
+Every system metric has a prefix ``system/`` when logged for grouping purpose. So the actual metric name gets logged
+are ``system/cpu_utilization_percentage``, ``system/system_memory_usage_megabytes``, etc.
+
+View System Metrics on MLflow UI
+--------------------------------
+
+System metrics are available on MLflow UI under the metrics section, let's start our MLflow UI server, and log system
+metrics to it:
+
+.. code-block:: bash
+
+    mlflow ui
+
+.. code-block:: python
+
+    import mlflow
+    import time
+
+    mlflow.set_tracking_uri("http://127.0.0.1:5000")
+    with mlflow.start_run() as run:
+        time.sleep(15)
+
+Navigate to `http://127.0.0.1:5000 <https://docs.databricks.com/en/machine-learning/index.html>`_ in your browser,
+and open your run, you should see system metrics under the metrics section, similar as shown by the screenshot below:
+
+.. figure:: ../_static/images/system-metrics/system-metrics-view.png
+    :width: 800px
+    :align: center
+    :alt: system metrics on MLflow UI
+
+
+Customize System Metrics Logging
+---------------------------------
+
+Customize Logging Frequency
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+By default system metrics are sampled every 10 seconds, and directly logged after sampling. You can customize
+the sampling frequency by setting environment variable ``MLFLOW_SYSTEM_METRICS_SAMPLING_INTERVAL`` or
+using :py:func:`mlflow.set_system_metrics_sampling_interval()`. You can also customize the number of samples to aggregate
+before logging by setting environment variable ``MLFLOW_SYSTEM_METRICS_SAMPLES_BEFORE_LOGGING`` or using
+:py:func:`mlflow.set_system_metrics_samples_before_logging()`. The actual logging time window is the product of
+``MLFLOW_SYSTEM_METRICS_SAMPLING_INTERVAL`` and ``MLFLOW_SYSTEM_METRICS_SAMPLES_BEFORE_LOGGING``. For example, if
+you set sample interval to 1 seconds and samples before logging to 3, then system metrics will be logged every 3s.
```

Comment:
Can some additional information be provided here for the explanation of the example? The reason it's 3s is that it's waiting for 3 samples and the metrics are being generated once per second. Perhaps using a more complex example that doesn't use a polling frequency (sample interval 2s, maybe?) to illustrate and explain the logic of how the sample frequency and aggregation count is used together could make this clear for users?

Reply:
makes sense, I added more explanation on this section, please take a look!